### PR TITLE
LastModified & Created Aware Mixins

### PR DESCRIPTION
**Pull request by nick-prat**

### CODE REVIEW

File: src/foam/nanos/auth/CreatedByAwareMixin.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'CreatedByAware',
+
+  implements: [
+    'foam.nanos.auth.CreatedByAware'
+  ],
+
+  properties: [
```
Comment (jlhughes) [inline src/foam/nanos/auth/CreatedByAwareMixin.js:15]: Can you also include the TableCellFormatters to show the user legalname?

File: src/foam/nanos/auth/CreatedByAwareMixin.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      updateVisibility: 'RO',
+      documentation: 'User who created the entry'
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.auth.User',
+      name: 'createdByAgent',
+      documentation: 'Agent acting as User who created the entry',
+      createVisibility: 'HIDDEN',
+      updateVisibility: 'RO',
```
Comment (kgrgreer) [inline src/foam/nanos/auth/CreatedByAwareMixin.js:30]: No trailing commas.

File: src/foam/nanos/auth/CreatedByAwareMixin.js (hunk 1 above)
Comment (kgrgreer) [reply src/foam/nanos/auth/CreatedByAwareMixin.js:15]: I think we could refine Reference itself to display the toSummary() of the referenced object, then we wouldn't have to do it here (or anywhere else).

Comment (kgrgreer) [reply src/foam/nanos/auth/CreatedByAwareMixin.js:15]: Actually, I don't think that was a good idea. In some cases, like a User's group, the group id is sufficient so no lookup or advanced formatter is required. We could however, make a generic ReferenceToSummaryFormatter which could be dropped in wherever we need it.

Comment (jlhughes) [reply src/foam/nanos/auth/CreatedByAwareMixin.js:15]: The formatter works, but group is the outlier. The majority of references are long or guid ids.

Comment (kgrgreer) [reply src/foam/nanos/auth/CreatedByAwareMixin.js:15]: I'm working on a generic Reference TableCellRenderer, which can be part of a future PR.

File: src/foam/nanos/auth/CreatedByAwareMixin.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'CreatedByAware',
```
Comment (Nauna) [inline src/foam/nanos/auth/CreatedByAwareMixin.js:9]: name: 'CreatedByAware',
  
  ...   name: 'CreatedByAwareMixin',

**Pull request merged**